camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/policy/GroupsChangePolicyOsgiTest.java

```diff
@@ -0,0 +1,472 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.camp.brooklyn.policy;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+import org.apache.brooklyn.api.effector.Effector;
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.api.entity.EntityLocal;
+import org.apache.brooklyn.api.mgmt.TaskAdaptable;
+import org.apache.brooklyn.camp.brooklyn.AbstractYamlTest;
+import org.apache.brooklyn.core.effector.CompositeEffector;
+import org.apache.brooklyn.core.effector.EffectorBody;
+import org.apache.brooklyn.core.effector.Effectors;
+import org.apache.brooklyn.core.entity.*;
+import org.apache.brooklyn.core.objs.AdjunctType;
+import org.apache.brooklyn.core.sensor.Sensors;
+import org.apache.brooklyn.core.sensor.StaticSensor;
+import org.apache.brooklyn.core.sensor.password.CreatePasswordSensor;
+import org.apache.brooklyn.core.test.entity.TestEntity;
+import org.apache.brooklyn.enricher.stock.Transformer;
+import org.apache.brooklyn.entity.group.DynamicGroup;
+import org.apache.brooklyn.entity.group.GroupsChangePolicy;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.apache.brooklyn.policy.InvokeEffectorOnSensorChange;
+import org.apache.brooklyn.test.Asserts;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.apache.brooklyn.test.Asserts.*;
+
+/**
+ *  Tests for adding {@link GroupsChangePolicy#INITIALIZERS}, {@link GroupsChangePolicy#LOCATIONS},
+ *  {@link GroupsChangePolicy#ENRICHERS} and {@link GroupsChangePolicy#POLICIES} to members once joined. Config keys and
+ *  attributes are expected to be resolved in the member context in all cases.
+ */
+public class GroupsChangePolicyOsgiTest extends AbstractYamlTest {
+
+    private ExecutorService executor;
+
+    @Override
+    @BeforeMethod(alwaysRun=true)
+    public void setUp() throws Exception {
+        super.setUp();
+        executor = Executors.newCachedThreadPool();
+    }
+
+
+    @AfterMethod(alwaysRun = true)
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        if (executor != null) executor.shutdownNow();
+    }
+
+    @Override
+    protected boolean disableOsgi() {
+        return false; // enable OSGI
```

Comment:
do we need this?  osgi tests are slower to set up and run so if not needed remove this bit (and rename the class)

Reply:
This is required for some tests, especially for `member.initializers` and `member.locaitons`.

Reply:
i think you are mistaken here

<img width="1441" alt="image" src="https://user-images.githubusercontent.com/496540/153401297-08a1c3f8-db04-48fb-908e-0b6e21dc6a89.png">

Reply:
You are right! I might have been getting that while I had old versions of tests, re-writing them to camp format resolved this 5626a6e95c8ff8d0a0743a36bcb5c60b9a244a9c.